src/reconcile/rec_visibility.c

```diff
@@ -388,6 +388,11 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
               vpack->start_ts <= upd_select->stop_ts && vpack->start_txn <= upd_select->stop_txn);
             upd_select->durable_ts = upd_select->start_ts = vpack->start_ts;
             upd_select->start_txn = vpack->start_txn;
+
+            /* Use stop timestamp as durable timestamp if exist. */
+            if (upd_select->stop_ts != WT_TS_MAX)
+                upd_select->durable_ts = upd_select->stop_ts;
+
```

Comment:
It isn't obvious to me that the update should be made durable at the same timestamp it was removed.  Can you explain this part, preferably in the comment above the code?

Reply:
I refactored this change to use the durable timestamp of the TOMBSTONE record. With this change, it uses the proper durable timestamp.